traversal/scanner/GraphIterator.java

```diff
@@ -344,6 +351,7 @@ private VertexMap toVertexMap(Map<Identifier, Vertex<?, ?>> answer) {
 
     @Override
     public void recycle() {
+        iterators.values().forEach(FunctionalIterator::recycle);
```

Comment:
which is what we do here - all opened iterators so far need to be recycled when recycle() is called.

Can't believe we didn't catch this before!